src/jdk.internal.le/share/classes/jdk/internal/org/jline/JdkConsoleProviderImpl.java

```diff
@@ -113,7 +113,8 @@ public String readLine() {
         public char[] readPassword(String fmt, Object ... args) {
             try {
                 initJLineIfNeeded();
-                return jline.readLine(fmt.formatted(args), '\0').toCharArray();
+                return jline.readLine(fmt.formatted(args).replace("%", "%%"), '\0')
```

Comment:
I find `\0` more cryptic and confusing than `(char) 0`, but okay. (Had to re-read https://docs.oracle.com/javase/specs/jls/se22/html/jls-3.html#jls-EscapeSequence)